### PR TITLE
Dinner (2/2)

### PR DESCRIPTION
**Pull request by amirahaile**
The instance variable `@soil_quality` creates a pretty dynamic orange grove. Based on the number of trees, the soil quality, orange production, and tree life span are all appropriately affected.

### CODE REVIEW

File: orange_tree.rb [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+      end
+
+      tree.one_year_passes
+
+      # checks again in case any trees died after a year passed
+      if tree.is_alive
+        trees_alive_this_year += 1
+      end
+
+      if trees_alive_this_year < trees_alive_last_year
```
Comment (davidmfoley) [inline orange_tree.rb:83]: It looks like `trees_alive_this_year` and `trees_alive_last_year` will not be set correctly each time this code is run.

File: orange_tree.rb [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+    end
+
+  end
+
+  def count_all_the_oranges
+    @trees.each do |tree|
+      @total_oranges += tree.orange_count
+    end
+
+    puts "The orange grove has #{@total_oranges} oranges."
```
Comment (davidmfoley) [inline orange_tree.rb]: The assignment specifies the count of oranges should be returned - this will return nil.

File: orange_tree.rb [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+      tree.one_year_passes
+
+      # checks again in case any trees died after a year passed
+      if tree.is_alive
+        trees_alive_this_year += 1
+      end
+
+      if trees_alive_this_year < trees_alive_last_year
+        # increase soil quality since providing for less trees
+        @soil_quality += 1
```
Comment (davidmfoley) [inline orange_tree.rb:85]: This happens for each tree (since it is inside the `@trees.each` loop) - is this correct?

File: orange_tree.rb [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+  end
+end
+
+class OrangeGrove
+  def initialize(num_of_trees)
+    @trees = []
+    @total_oranges = 0
+    @soil_quality = 50 # 0 = low soil quality; 50 = high soil quality
+
+    num_of_trees.times do
```
Comment (davidmfoley) [inline orange_tree.rb:60]: fancy!

File: orange_tree.rb (hunk 1 above)
Comment (amirahaile) [inline orange_tree.rb:83]: How do you mean? It seems to be accurate when I test… I simply want to check if the state of `is_alive` changed for any of my `OrangeTree` instances after applying `one_year_passes`.

File: orange_tree.rb (hunk 3 above)
Comment (amirahaile) [inline orange_tree.rb:85]: Yes, the soil quality of the grove should be affected based on the number of trees in the grove. So, more trees == poor soil quality and vice versa.

File: orange_tree.rb (hunk 1 above)
Comment (davidmfoley) [inline orange_tree.rb:83]: Let's say there are 5 trees that are alive, `tree[0]...tree[4]`:

If the first one dies (`tree[0]`): 
on each iteration of the loop, `trees_alive_this_year < trees_alive_last_year` will be true - so `@soil_quality` will be decremented 5 times (net change -5). 

On the other hand if the last one (`tree[4]`) dies:
`trees_alive_this_year < trees_alive_last_year` will only be true on the last iteration, and @soil_quality will be incremented four times and decremented only once (for a net +3).

**Pull request closed without merging**